configure-slic/action.yml

```diff
@@ -0,0 +1,71 @@
+name: Configure slic
+description: Installs and configures slic
+
+inputs:
+  wp_version:
+    required: false
+    default: ''
+    description: WordPress version to be installed
+  airplane_mode:
+    required: false
+    default: 'on'
+    description: Airplane mode, no external connections (on/off)
+
+runs:
+  using: composite
+  steps:
+    - name: Checkout slic
+      uses: actions/checkout@v4
+      with:
+        repository: stellarwp/slic
+        ref: main
+        path: slic
+        fetch-depth: 1
+
+    - name: Set up slic env vars
+      shell: bash
+      run: |
+        echo "SLIC_BIN=${GITHUB_WORKSPACE}/slic/slic" >> $GITHUB_ENV
+        echo "SLIC_WP_DIR=${GITHUB_WORKSPACE}/slic/_wordpress" >> $GITHUB_ENV
+        echo "SLIC_WORDPRESS_DOCKERFILE=Dockerfile.base" >> $GITHUB_ENV
+
+    - name: Set run context for slic
+      shell: bash
+      run: echo "SLIC=1" >> $GITHUB_ENV && echo "CI=1" >> $GITHUB_ENV
+
+    - name: Start ssh-agent
+      shell: bash
+      run: |
+        mkdir -p "${HOME}/.ssh";
+        ssh-agent -a /tmp/ssh_agent.sock;
+
+    - name: Export SSH_AUTH_SOCK env var
+      shell: bash
+      run: echo "SSH_AUTH_SOCK=/tmp/ssh_agent.sock" >> $GITHUB_ENV
+
+    - name: Set up slic for CI
+      shell: bash
+      run: |
+        cd ${GITHUB_WORKSPACE}/..
+        ${SLIC_BIN} here
+        ${SLIC_BIN} interactive off
+        ${SLIC_BIN} build-prompt off
+        ${SLIC_BIN} build-subdir off
+        ${SLIC_BIN} xdebug off
+        ${SLIC_BIN} debug on
```

Comment:
We should add another input here for the PHP version for slic, defaulting to `7.4`, e.g.

```
${SLIC_BIN} php-version set 8.1 --skip-rebuild
```

Reply:
`slic.json` definitely won't work, because if anything forces slic to load the containers before the `slic use plugin-slug` command (e.g. any `wp` command), it will not load the correct PHP version and will cause unneeded container downloads.

I can't remember if `.env.slic.local` will properly do this, would you have time to test? 

The main thing we want to avoid is slic downloading all the 7.4 containers when you plan to use 8.1, for example, as that is usually one of the longer running tasks.

Reply:
Actually, `php-version set` writes to env file https://github.com/stellarwp/slic/blob/fe761e103f00df823cb4e83c383c54828a770ae1/src/slic.php#L324. So, the env file is the single source of truth.

Reply:
The problem with that, is it's not loaded until a `slic use plugin-slug` has been run, otherwise it doesn't know its target path, so the `wp core update` command would start downloading the `7.4` containers even if you wanted to use `8.1` as the `slic use` command doesn't happen until way later in the execution.

https://github.com/stellarwp/slic/blob/main/src/project.php#L21
https://github.com/stellarwp/slic/blob/main/src/slic.php#L355

I'm pretty sure we need an input and to set it here before the containers have any chance to spin up.